classes/Carrier.php

```diff
@@ -1401,7 +1407,7 @@ public static function getAvailableCarrierList(Product $product, $id_warehouse,
                     unset($carrier_list[$key]);
                 }
 
-                if ($carrier->max_weight > 0 && $carrier->max_weight < $product->weight * $cart_quantity) {
+                if ($carrier->max_weight > 0 && ($carrier->max_weight < $product->weight * $cart_quantity OR $carrier->max_weight < $cart_weight)) {
```

Comment:
As mentionned here : http://forge.prestashop.com/browse/PSCSX-6751?focusedCommentId=156257&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-156257

Shouldn't this condition become only "if ($carrier->max_weight > 0 && $carrier->max_weight < $cart_weight)" ?